Merge/foundation release/1.12.2

ziogaschr opened the pull request:
Merges [ethereum/go-ethereum branch release/1.12.2](https://github.com/ethereum/go-ethereum/releases/tag/v1.12.2) to core-geth master.

# Notes
---
- Do you agree on replacing Clique with Catalyst for `--dev` mode? We still allow use `--dev.pow` for ethash. 9d4ea2e83fc21b41135984363a06334aff106055
- This change follows the go-ethereum interfaces, where a Block header can have `withdrawalsRoot` while the Block itself can have `withdrawals`.  23da7ca9d6ae460324a766a0830b4275b8dc9764
...

# TODO

- [x] Re-implement `RemovePendingTransaction`, as it's not working now that blobpool has been added Ref: https://github.com/etclabscore/core-geth/pull/203
    - :arrow_right: We decided to remove this method since it's not well supported by all transaction types anymore. (https://github.com/etclabscore/core-geth/pull/560#discussion_r1311523912)
- [x] Test that `--dev` mode works https://github.com/etclabscore/core-geth/pull/560#issuecomment-1698943446

stephenfire committed:
rlp: use identical receiver names for encBuffer methods (#27430)

ThreeAndTwo committed:
accounts/keystore: handle error for invalid key in DecryptKey (#27432)


Co-authored-by: KAI <kaili@coinsummer.io>
Co-authored-by: Felix Lange <fjl@twurst.com>

jsvisa committed:
eth/filters: reuse handleLogs method for removed logs (#27438)

jwasinger committed:
miner: log number of withdrawals in block upon payload update (#27457)

hero5512 committed:
internal/ethapi: remove error return on RPCMarshalBlock (#27449)

rm error when marshal block to rpc type allen

mmsqe committed:
rpc: add limit for batch request items and response size (#26681)

This PR adds server-side limits for JSON-RPC batch requests. Before this change, batches
were limited only by processing time. The server would pick calls from the batch and
answer them until the response timeout occurred, then stop processing the remaining batch
items.

Here, we are adding two additional limits which can be configured:

- the 'item limit': batches can have at most N items
- the 'response size limit': batches can contain at most X response bytes

These limits are optional in package rpc. In Geth, we set a default limit of 1000 items
and 25MB response size.

When a batch goes over the limit, an error response is returned to the client. However,
doing this correctly isn't always possible. In JSON-RPC, only method calls with a valid
`id` can be responded to. Since batches may also contain non-call messages or
notifications, the best effort thing we can do to report an error with the batch itself is
reporting the limit violation as an error for the first method call in the batch. If a batch is
too large, but contains only notifications and responses, the error will be reported with
a null `id`.

The RPC client was also changed so it can deal with errors resulting from too large
batches. An older client connected to the server code in this PR could get stuck
until the request timeout occurred when the batch is too large. **Upgrading to a version
of the RPC client containing this change is strongly recommended to avoid timeout issues.**

For some weird reason, when writing the original client implementation, @fjl worked off of
the assumption that responses could be distributed across batches arbitrarily. So for a
batch request containing requests `[A B C]`, the server could respond with `[A B C]` but
also with `[A B] [C]` or even `[A] [B] [C]` and it wouldn't make a difference to the
client.

So in the implementation of BatchCallContext, the client waited for all requests in the
batch individually. If the server didn't respond to some of the requests in the batch, the
client would eventually just time out (if a context was used).

With the addition of batch limits into the server, we anticipate that people will hit this
kind of error way more often. To handle this properly, the client now waits for a single
response batch and expects it to contain all responses to the requests.

---------

Co-authored-by: Felix Lange <fjl@twurst.com>
Co-authored-by: Martin Holst Swende <martin@swende.se>

freeman-jiang committed:
core/types: add "chainID" field to legacy tx JSON encoding (#27452)


Co-authored-by: lightclient <14004106+lightclient@users.noreply.github.com>

zhiqiangxu committed:
rpc: add method to test for subscription support (#25942)

This adds two ways to check for subscription support. First, one can now check
whether the transport method (HTTP/WS/etc.) is capable of subscriptions using
the new Client.SupportsSubscriptions method.

Second, the error returned by Subscribe can now reliably be tested using this
pattern:
    
    sub, err := client.Subscribe(...)
    if errors.Is(err, rpc.ErrNotificationsUnsupported) {
        // no subscription support
    }

---------

Co-authored-by: Felix Lange <fjl@twurst.com>

pengin7384 committed:
node: remove unused error return from Attach (#27450)

node: Delete the unused error from return parameters of Node.Attach() func

s1na committed:
eth/tracers: refactor exporting js buffer (#27472)

We had to do this workaround because it wasn't possible to export typed arrays from
JS to []byte. This was added in dop251/goja@2352993, so we can use the better way now.

aaronbuchwald committed:
core: remove unnecessary log copy (#27475)

The logs in this function are pulled straight from disk in rawdb.ReadRawReceipts and 
also modified in receipts.DeriveFields, so removing the copy should be fine.

holiman committed:
core/types: add support for BlobTxType receipts (#27470)


Co-authored-by: Felix Lange <fjl@twurst.com>

cuishuang committed:
cmd/evm: fix typos in docs (#27478)

fix some typos

Signed-off-by: cui fliter <imcusg@gmail.com>

joohhnnn committed:
rlp: fix typos (#27484)

fix typos

Co-authored-by: john <yejiarui@123.com>

karalabe committed:
all: move main transaction pool into a subpool (#27463)

* all: move main transaction pool into a subpool

* go.mod: remove superfluous updates

* core/txpool: review fixes, handle txs rejected by all subpools

* core/txpool: typos

Commit:
build: use slices package for sorting (#27486)

lochjin committed:
core/txpool/legacypool: reheap the tx list if london fork not enabled (#27481)

This change ensures Reheap will be called even before the London fork activates.
Since Reheap would otherwise only be called through `SetBaseFee` after London,
the list would just keep growing if the fork was not enabled or not reached yet.

Commit:
p2p: use slices package for sorting (#27494)


Co-authored-by: Felix Lange <fjl@twurst.com>

Commit:
ethdb/dbtest: use slices package for sorting (#27491)

joohhnnn committed:
internal/jsre/deps: fix typos in web3.js (#27500)

hero5512 committed:
core/txpool/legacypool: remove redundant check for floatingRatio == 0 (#27477)

floatingRatio is a constant and always non-zero. So there is no need to
check for == 0.

holiman committed:
core/txpool/legacypool: handle missing head in reset (#27479)

Fixes #27301, a crash that could occur during txpool reorg handling.

zchee committed:
rpc: avoid use of cgo by hard-coding maxPathSize (#27447)

Package rpc uses cgo to find the maximum UNIX domain socket path 
length. If exceeded, a warning is printed. This is the only use of cgo in this
package. It seems excessive to depend on cgo just for this warning, so
we now hard-code the usual limit for Linux instead.

---------

Co-authored-by: Felix Lange <fjl@twurst.com>

Commit:
accounts/keystore: use slices package for sorting (#27485)


Co-authored-by: Felix Lange <fjl@twurst.com>

Commit:
cmd/devp2p: use slices package for sorting (#27487)

joohhnnn committed:
cmd/evm: fix typos in documentation (#27501)

pengin7384 committed:
consensus/clique: fix typo in comment (#27503)

Commit:
metrics: use slices package for sorting (#27493)


Co-authored-by: Felix Lange <fjl@twurst.com>

Commit:
consensus/clique: use slices package for sorting (#27488)

Also adds Address.Less for sorting use in other packages.

---------

Co-authored-by: Felix Lange <fjl@twurst.com>

Commit:
internal/ethapi, les: use slices package for sorting (#27492)


Co-authored-by: Felix Lange <fjl@twurst.com>

Commit:
tests, trie: use slices package for sorting (#27496)


Co-authored-by: Felix Lange <fjl@twurst.com>

MariusVanDerWijden committed:
go.mod: update kzg libraries to use big-endian (#27510)

* go.mod: update kzg libraries to use big-endian

* go.sum: ran go mod tidy

* core/testdata/precompiles: fix blob verification test

* core/testdata/precompiles: fix blob verification test

rjl493456442 committed:
trie: remove parameter 'fromLevel' in Prove (#27512)

This removes the feature where top nodes of the proof can be elided.
It was intended to be used by the LES server, to save bandwidth 
when the client had already fetched parts of the state and only needed
some extra nodes to complete the proof. Alas, it never got implemented
in the client.

hero5512 committed:
core/txpool/legacypool: narrow down the scope of the variable (#27471)

Variables discarded, included can be defined closer to their usage.

Co-authored-by: Martin Holst Swende <martin@swende.se>

Commit:
eth: use slices package for sorting (#27490)

Also adds Hash.Less method for sorting purposes.

---------

Co-authored-by: Felix Lange <fjl@twurst.com>

jsvisa committed:
internal/build: ignore some files in FindMainPackages (#27521)

Commit:
core: use slices package for sorting (#27489)


Co-authored-by: Felix Lange <fjl@twurst.com>

fjl committed:
core/txpool: remove use of errors.Join function (#27523)

his function was added in Go 1.20, but our compatibility target
is Go 1.19.

jwasinger committed:
internal/ethapi: use same state for each invocation within EstimateGas (#27505)

EstimateGas repeatedly executes a transaction, performing a binary search with multiple gas prices to determine proper pricing. Each call retrieves a new copy of the state (https://github.com/ethereum/go-ethereum/blob/master/internal/ethapi/api.go#L1017) . Because the pending/latest state can change during the execution of EstimateGas, this can potentially cause strange behavior (as noted here: https://github.com/ethereum/go-ethereum/pull/27502#issue-1761957009).

This PR modifies EstimateGas to retrieve the state once and use a copy of it for every call invocation it does.

rjl493456442 committed:
cmd, core/state, eth, tests, trie: improve state reader (#27428)

The state availability is checked during the creation of a state reader.

-    In hash-based database, if the specified root node does not exist on disk disk, then
    the state reader won't be created and an error will be returned.

-    In path-based database, if the specified state layer is not available, then the
    state reader won't be created and an error will be returned.

This change also contains a stricter semantics regarding the `Commit` operation: once it has been performed, the trie is no longer usable, and certain operations will return an error.

holiman committed:
ethdb/pebble: fsync for batch writes (#27522)

This is likely the culprit behind several data corruption issues, e.g. where data has been
written to the freezer, but the deletion from pebble does not go through due to process
crash.

prestwich committed:
internal/ethapi: quantity-encode storage keys in eth_getProof response (#27309)

This changes the eth_getProof method implementation to re-encode the requested
storage keys, canonicalizing them in the response. For backwards-compatibility reasons,
go-ethereum accepts non-canonical hex keys. Accepting them is fine, but we should
not mirror invalid inputs into the output.

Closes #27306

---------

Co-authored-by: Martin Holst Swende <martin@swende.se>
Co-authored-by: Felix Lange <fjl@twurst.com>

Exca-DK committed:
p2p/discover: swap verification order in discv4 ping handler (#27532)

In all other UDPv4 methods, the deadline is checked first. It seems weird to me that ping is an exception. Deadline comparison is also less resource intensive.

Co-authored-by: Exca-DK <Exca-DK@users.noreply.github.com>

gballet committed:
core/state, light, trie: add UpdateContractCode to the Trie interface (#27476)

Verkle trees store the code inside the trie. This PR changes the interface to pass the code, as well as the dirty flag to tell the trie package if the code is dirty and needs to be updated. This is a no-op for the MPT and the odr trie.

fbac committed:
crypto/secp256k1: define NDEBUG only if not defined (#27550)

pengin7384 committed:
common/math: fix typo in comment (#27561)

jsvisa committed:
eth/protocols: put protocols in order (#27543)

Signed-off-by: jsvisa <delweng@gmail.com>

puhtaytow committed:
cmd/geth: error message should not be capitalised (#27549)

error message should not be capitalized / consistency

rjl493456442 committed:
ethdb/pebble: use sync mode for pebble writes (#27615)

MariusVanDerWijden committed:
core/vm: move TSTORE,TLOAD to correct opcode nums (#27613)

* core/vm: move TSTORE,TLOAD to correct opcode nums

* core/vm: cleanup

pengin7384 committed:
common/lru: add test case for BasicLRU.Peek (#27559)

aaronbuchwald committed:
core: log message if DeriveFields returns error (#27530)

This error will never happen, but if it ever does happen, we will at least see it.

gballet committed:
trie, core/state: revert error removal in (*state.Trie).Commit (#27544)

* trie, core/state: revert error removal in (*state.Trie).Commit

* Gary's nitpick :)

Co-Authored-By:  rjl493456442 <garyrong0905@gmail.com>

---------

Co-authored-by: rjl493456442 <garyrong0905@gmail.com>

hero5512 committed:
core/txpool/legacypool: fix typo (#27620)

MariusVanDerWijden committed:
eth: send big transactions by announce/retrieve only (#27618)

* eth: send big transactions by announce/retrieve only

* Update eth/handler.go

Co-authored-by: Martin Holst Swende <martin@swende.se>

* eth: remove superfluous bracket

* eth: add whitespace

---------

Co-authored-by: Martin Holst Swende <martin@swende.se>

gballet committed:
params, trie: add verkle fork management + upgrade go-verkle (#27464)

* params, trie: add verkle fork management + upgrade go-verkle

* remove the two verkle files

* core, eth, params: add missing function

* Gary's feedback

* remove trie/utils/verkle.go

* add verkle block override

---------

Co-authored-by: Gary Rong <garyrong0905@gmail.com>

gballet committed:
core/state, light, les: make signature of ContractCode hash-independent (#27209)

* core/state, light, les: make signature of ContractCode hash-independent

* push current state for feedback

* les: fix unit test

* core, les, light: fix les unittests

* core/state, trie, les, light: fix state iterator

* core, les: address comments

* les: fix lint

---------

Co-authored-by: Gary Rong <garyrong0905@gmail.com>

oiooj committed:
node: fix listening on IPv6 address (#27628) (#27635)

rjl493456442 committed:
cmd, core, eth, graphql, trie: no persisted clean trie cache file (#27525)

The clean trie cache is persisted periodically, therefore Geth can
quickly warmup the cache in next restart.

However it will reduce the robustness of system. The assumption is
held in Geth that if the parent trie node is present, then the entire
sub-trie associated with the parent are all prensent.

Imagine the scenario that Geth rewinds itself to a past block and
restart, but Geth finds the root node of "future state" in clean
cache then regard this state is present in disk, while is not in fact.

Another example is offline pruning tool. Whenever an offline pruning
is performed, the clean cache file has to be removed to aviod hitting
the root node of "deleted states" in clean cache.

All in all, compare with the minor performance gain, system robustness
is something we care more.

Exca-DK committed:
ethdb/leveldb: update leveldb metrics collection (#27643)

This removes text parsing in leveldb metrics collection code. All metrics
can now be accessed through the stats API provided by leveldb.

We also add new gauge-typed metrics that count the number of tables at each level.

---------

Co-authored-by: Exca-DK <Exca-DK@users.noreply.github.com>
Co-authored-by: Gary Rong <garyrong0905@gmail.com>
Co-authored-by: Felix Lange <fjl@twurst.com>

Commit:
core: stop chain when tests are finished (#27660)

core (blockchain_test): add chain.Stop() to tests

joohhnnn committed:
signer/core: show helptext when clef rejects signing (#27538)



Co-authored-by: Martin Holst Swende <martin@swende.se>

jwasinger committed:
cmd/geth: implement dev mode for post-merge (#27327)

This change adds back the 'geth --dev' mode of operation, using a cl-mocker. 

---------

Co-authored-by: Martin Holst Swende <martin@swende.se>
Co-authored-by: rjl493456442 <garyrong0905@gmail.com>
Co-authored-by: lightclient <14004106+lightclient@users.noreply.github.com>

lightclient committed:
p2p, p2p/discover: add dial metrics (#27621)

This PR adds metrics for p2p dialing, which gives us visibility into the quality of the dial 
candidates  returned by our discovery methods.

holiman committed:
core: remove outdated tests (#27662)

Back before #27178 , we spun up a number of ethash verifiers to verify headers. So we also had tests to ensure that we were indeed able to abort verification even if we had multiple workers running.

With PR #27178, we removed the parallelism in verification, and these tests are now failing, since we now just sequentially fire away the results as fast as possible on one routine.

This change removes the (sometimes failing) tests

jwasinger committed:
miner: remove unused commented code (#27664)

holiman committed:
core/vm: define cancun + enable 1153 (tstore/tload) in cancun (#27663)

charles-cooper committed:
core/vm: implement EIP-5656, mcopy instruction (#26181)

Implements [EIP 5656](https://eips.ethereum.org/EIPS/eip-5656), MCOPY instruction, and enables it for Cancun. 

---------

Co-authored-by: Martin Holst Swende <martin@swende.se>

lightclient committed:
eth: better active protocol handler tracking (#27665)

Fixes an issue where waitgroups were used erroneously, which could lead to waitgroup being added to while wait was already invoked.

jsvisa committed:
internal/blocktest: add package for shared test code (#27270)

rjl493456442 committed:
core/state: value diff tracking in StateDB (#27349)

This change makes the StateDB track the state key value diff of a block transition.
We already tracked current account and storage values for the purpose of updating
the state snapshot. With this PR, we now also track the original (pre-transition) values
of accounts and storage slots.

lightclient committed:
cmd/utils, p2p: clean up discovery setup (#27518)

This simplifies the code that initializes the discovery a bit, and
adds new flags for enabling/disabling discv4 and discv5 separately.

---------

Co-authored-by: Felix Lange <fjl@twurst.com>

Mskxn committed:
console: add cleanup to avoid leaks in newTester (#27695)

fjl committed:
go.mod: upgrade github.com/karalabe/usb to fix build warning (#27698)

rjl493456442 committed:
core/rawdb: support freezer batch read with no size limit (#27687)

This change adds the ability to perform reads from freezer without size limitation. This can be useful in cases where callers are certain that out-of-memory will not happen (e.g. reading only a few elements). 

The previous API was designed to behave both optimally and secure while servicing a request from a peer, whereas this change should _not_ be used when an untrusted peer can influence the query size.

fjl committed:
p2p/discover: filter bootnodes by netrestrict (#27701)

This prevents an issue where the node would attempt to contact the bootstrap
nodes even if they weren't contained in the netrestrict list.

cuishuang committed:
accounts, cmd/evm: fix docstrings (#27703)

fix function name in comment

Signed-off-by: cui fliter <imcusg@gmail.com>

sandakersmann committed:
build: upgrade -dlgo version to Go 1.20.6 (#27708)

jsvisa committed:
eth, miner: rm redundant type declare (#27713)

* eth: rm redundant type from array, slice, or map

* miner: rm redundant type from array, slice, or map

s1na committed:
graphql: fix race in withdrawals test (#27706)

graphql: fix race in test

jsvisa committed:
internal: early exit if tx for getTxReceipt not found (#27712)

internal/ethapi: fast exit if tx notfound

jsvisa committed:
internal/ethapi: add more testcase for GetTransactionReceipt (#27527)

* internal/ethapi: testBackend reuse the same db

Signed-off-by: jsvisa <delweng@gmail.com>

* internal/ethapi: implment GetTransaction

Signed-off-by: jsvisa <delweng@gmail.com>

* internal/ethapi: implement GetReceipts

Signed-off-by: jsvisa <delweng@gmail.com>

* internal/ethapi: insert receipts and setup txlookup

Signed-off-by: jsvisa <delweng@gmail.com>

* internal/ethapi: add simple success tx receipt test

Signed-off-by: jsvisa <delweng@gmail.com>

* internal/ethapi: add case create contract

Signed-off-by: jsvisa <delweng@gmail.com>

* internal/ethapi: add contract call receipt

Signed-off-by: jsvisa <delweng@gmail.com>

* internal/ethapi: add tx notfound

Signed-off-by: jsvisa <delweng@gmail.com>

* internal/ethapi: add dynamic fee testcase

* internal/ethapi: add accessList receipt

* internal/ethapi: no need to insert receipt chain, no error

* internal/ethapi: use HeaderByHash instead

* internal/ethapi: add one more case

---------

Signed-off-by: jsvisa <delweng@gmail.com>

jsvisa committed:
eth/tracers: add ReturnData in the tracer's response (#27704)

The struct logger (or opcode tracer) was missing the return data field even
if this was explicitly enabled by user via `"enableReturnData": true` in the config.
This PR fixes this issue.

Co-authored-by: Sina Mahmoodi <itz.s1na@gmail.com>

ryanschneider committed:
eth/tracers/native: panic on memory read in prestateTracer (#27691)


Co-authored-by: Sina Mahmoodi <itz.s1na@gmail.com>

ucwong committed:
go.sum: go mod tidy (#27717)

ucwong committed:
core/blockchain: fast to snap in comments (#27722)

dbadoy committed:
cmd/bootnode, p2p: support for alternate mapped ports (#26359)

This changes the port mapping procedure such that, when the requested port is unavailable
an alternative port suggested by the router is used instead.

We now also repeatedly request the external IP from the router in order to catch any IP changes.

Co-authored-by: Felix Lange <fjl@twurst.com>

omerfirmak committed:
metrics: NilTimer should still run the function to be timed (#27723)

jsvisa committed:
cmd/geth: use automaxprocs to apply cpu quota correctly (#27506)

It is usually best to set GOMAXPROCS to the number of available CPU cores. However, setting
it like that does not work well when the process is quota-limited to a certain number of CPUs.
The automaxprocs library configures GOMAXPROCS, taking such limits into account.

omerfirmak committed:
metrics: NilResettingTimer.Time should execute the timed function (#27724)

yuriechan committed:
accounts/abi/bind/backends: fix goroutine leak in unit test (#27705)

jwasinger committed:
core:  replace instances of 'suicide' with 'selfdestruct' to improve code consistency. (#27716)


---------

Co-authored-by: Martin Holst Swende <martin@swende.se>
Co-authored-by: lightclient <14004106+lightclient@users.noreply.github.com>

ucwong committed:
core/vm: use types.EmptyCodeHash (#27729)

holiman committed:
core, tests: EIP-4844 transaction processing logic (#27721)

This updates the reference tests to the latest version and also adds logic
to process EIP-4844 blob transactions into the state transition. We are now
passing most Cancun fork tests.

Co-authored-by: Marius van der Wijden <m.vanderwijden@live.de>
Co-authored-by: Felix Lange <fjl@twurst.com>

jwasinger committed:
core/state, core/vm: implement EIP 6780 (#27189)

EIP-6780: SELFDESTRUCT only in same transaction

>     SELFDESTRUCT will recover all funds to the caller but not delete the account, except when called in the same transaction as creation

---------

Co-authored-by: Martin Holst Swende <martin@swende.se>

zsfelfoldi committed:
les: remove obsolete code related to PoW header syncing (#27737)

This change removes PoW header syncing related code from LES and also deletes 
duplicated packages les/catalyst, les/downloader and les/fetcher. These package copies
were created because people wanted to make changes in their eth/ counterparts, but weren't
able to adapt LES code to the API changes.

fjl committed:
beacon/engine, eth/catalyst: EIP-4844 updates for the engine API (#27736)

This is a spin-out from the EIP-4844 devnet branch, containing just the Engine API modifications
and nothing else. The newPayloadV3 endpoint won't really work in this version, but we need the
data structures for testing so I'd like to get this in early.

Co-authored-by: Marius van der Wijden <m.vanderwijden@live.de>

pgrimaud committed:
cmd/evm/testdata: fix typos in docs (#27742)

MariusVanDerWijden committed:
ethclient/gethclient: gofmt -s (#27762)

rjl493456442 committed:
all: expose block number information to statedb (#27753)

* core/state: clean up

* all: add block number infomration to statedb

* core, trie: rename blockNumber to block

ucwong committed:
cmd/geth: atomic types used (#27756)

ucwong committed:
cmd/devp2p: atomic types used (#27755)

jtraglia committed:
crypto/kzg4844: do lazy init in all ckzg funcs (#27679)

* crypto/kzg4844: remove unnecessary init call & fix typo

* Fix kzg4844 tests/benchmarks

* Make init lazy & revert changes to tests

Commit:
core/types: deepcopy ExcessDataGas, DataGasUsed (#27767)

karalabe committed:
core/txpool/blobpool: 4844 blob transaction pool (#26940)

* core/blobpool: implement txpool for blob txs

* core/txpool: track address reservations to notice any weird bugs

* core/txpool/blobpool: add support for in-memory operation for tests

* core/txpool/blobpool: fix heap updating after SetGasTip if account is evicted

* core/txpool/blobpool: fix eviction order if cheap leading txs are included

* core/txpool/blobpool: add note as to why the eviction fields are not inited in reinject

* go.mod: pull in inmem billy form upstream

* core/txpool/blobpool: fix review commens

* core/txpool/blobpool: make heap and heap test deterministic

* core/txpool/blobpool: luv u linter

* core/txpool: limit blob transactions to 16 per account

* core/txpool/blobpool: fix rebase errors

* core/txpool/blobpool: luv you linter

* go.mod: revert some strange crypto package dep updates

fjl committed:
core/types: add 4844 data gas fields in Receipt (#27743)

* core/types: add data gas fields in Receipt

* core/types: use BlobGas method of tx

* core: fix test

* core/types: fix receipt tests, add data gas used field test

---------

Co-authored-by: Péter Szilágyi <peterke@gmail.com>

karalabe committed:
core/txpool/blobpool: fix a merge conflict from a package rename (#27790)

karalabe committed:
all: rename dataGas to blobGas (#27789)

MariusVanDerWijden committed:
core/types: put header fields in correct order (#27791)

marioevz committed:
core/types: fix receipt blob fields marshaling (#27793)

islishude committed:
cmd/geth: import package catalyst once (#27803)

islishude committed:
eth: conver if-else-if chain to tagged switch (#27816)

jsvisa committed:
core: remove pointless check for excessBlobGas nilness (#27797)

* core: check excessBlobGas in front

Signed-off-by: jsvisa <delweng@gmail.com>

* core: no need to manual panic

Signed-off-by: jsvisa <delweng@gmail.com>

* core: no comment

Signed-off-by: jsvisa <delweng@gmail.com>

---------

Signed-off-by: jsvisa <delweng@gmail.com>

rjl493456442 committed:
core, trie: track state change set with account address (#27815)

ZigBalthazar committed:
cmd/bootnode: defer udp connection close (#27787)

jsvisa committed:
all: replace data gas to blob gas in comments (#27825)

* eth: excessDataGas -> excessBlobGas

* consensus: data gas -> blob gas

* core: data gas -> blob gas

* params: data gas -> blob gas

holiman committed:
tests: make tests pass on windows/386 (#27821)

* tests: split up state test execution

* Revert "tests: split up state test execution"

This reverts commit 96017c248c85d24e93ad013a2bbe8b38c99327c0.

* build: bump test timeout to 20 minutes

rjl493456442 committed:
core/txpool/blobpool: remove error log when finalized block is nil (#27822)

* core/txpool/blobpool: remove error log when finalized block is nil

* core/txpool/blobpool: take peter's suggestion

jsvisa committed:
consensus/misc: move eip1559 into a package (#27828)

* consensus/misc: move eip1559 as a sub directory

Signed-off-by: jsvisa <delweng@gmail.com>

* consensus/misc: package name

Signed-off-by: jsvisa <delweng@gmail.com>

* all: eip1559

Signed-off-by: jsvisa <delweng@gmail.com>

---------

Signed-off-by: jsvisa <delweng@gmail.com>

jsvisa committed:
cmd/evm: set ExcessBlobGas from env (#27796)

Sets the `currentExcessBlobGas` from env, alternatively calculates it based on `parentExcessBlobGas` and `parentBlobGasUsed`. It then emits the `currentExcessBlobGas` and `currentBlobGasUsed` into the output, to be used as parent-values for a future iteration. 

Closes #27785
Closes #27783

---------

Signed-off-by: jsvisa <delweng@gmail.com>
Co-authored-by: Martin Holst Swende <martin@swende.se>

ucwong committed:
cmd/bootnode: fix timer leak (#27754)

rjl493456442 committed:
all: implement path-based state scheme (#25963)

* all: implement path-based state scheme

* all: edits from review

* core/rawdb, trie/triedb/pathdb: review changes

* core, light, trie, eth, tests: reimplement pbss history

* core, trie/triedb/pathdb: track block number in state history

* trie/triedb/pathdb: add history documentation

* core, trie/triedb/pathdb: address comments from Peter's review

Important changes to list:

- Cache trie nodes by path in clean cache
- Remove root->id mappings when history is truncated

* trie/triedb/pathdb: fallback to disk if unexpect node in clean cache

* core/rawdb: fix tests

* trie/triedb/pathdb: rename metrics, change clean cache key

* trie/triedb: manage the clean cache inside of disk layer

* trie/triedb/pathdb: move journal function

* trie/triedb/path: fix tests

* trie/triedb/pathdb: fix journal

* trie/triedb/pathdb: fix history

* trie/triedb/pathdb: try to fix tests on windows

* core, trie: address comments

* trie/triedb/pathdb: fix test issues

---------

Co-authored-by: Felix Lange <fjl@twurst.com>
Co-authored-by: Martin Holst Swende <martin@swende.se>

jsvisa committed:
cmd/geth: disable automaxprocs log (#27814)


Co-authored-by: Felix Lange <fjl@twurst.com>

ucwong committed:
p2p: use atomic types (#27764)


Co-authored-by: Felix Lange <fjl@twurst.com>

sandakersmann committed:
build: upgrade -dlgo version to Go 1.20.7 (#27835)

karalabe committed:
eth/protocols/snap: fix batch writer when resuming an aborted sync (#27842)

fjl committed:
core/types: fix immutability guarantees in Block (#27844)

This change rearranges the accessor methods in block.go and fixes some minor issues with
the copy-on-write logic of block data. Fixed issues:

- Block.WithWithdrawals did not create a shallow copy of the block.

- Block.WithBody copied the header unnecessarily, and did not preserve withdrawals.

However, the bugs did not affect any code in go-ethereum because blocks are *always*
created using NewBlockWithHeader().WithBody().WithWithdrawals()

Commit:
all: remove trailing whitespace (#27741)

fjl committed:
core/types: support yParity field in JSON transactions (#27744)

This adds support for the "yParity" field in transaction objects returned by RPC
APIs. We somehow forgot to add this field even though it has been in the spec for
a long time.

ucwong committed:
log: use atomic types (#27763)


Co-authored-by: Felix Lange <fjl@twurst.com>

chrischarlesharrison committed:
internal/ethapi, graphql: correct comments about gas price logic (#27752)

ucwong committed:
internal: use atomic type (#27858)

ucwong committed:
accounts: use atomic type (#27857)

amintalebi committed:
internal/ethapi: add state override to estimateGas (#27845)

ucwong committed:
p2p/simulations/examples: use atomic.Int64 (#27861)

holiman committed:
graphql: avoid greedy allocation (#27873)

Fixes a graphql-dos

---------

Co-authored-by: Sina Mahmoodi <1591639+s1na@users.noreply.github.com>
Co-authored-by: Sina Mahmoodi <itz.s1na@gmail.com>

holiman committed:
core/vm: update 4844 - point evaluation precompile address (#27874)

fjl committed:
p2p: move ping handling into pingLoop goroutine (#27887)

Moving the response sending there allows tracking all peer goroutines
in the peer WaitGroup.

jsvisa committed:
graphql: add yParity field for transactions (#27882)


Co-authored-by: Felix Lange <fjl@twurst.com>

gballet committed:
core/state: move UpdateContractCode before the trie hash is computed (#27853)

Context: The UpdateContractCode method was introduced for the state storage commitment
schemes that include the whole code for their commitment computation. It must therefore be called
before the root hash is computed at the end of IntermediateRoot.

This should have no impact on the MPT since, in this context, the method is a no-op.

fjl committed:
params: release go-ethereum v1.12.1

fjl committed:
params: begin v1.12.2 release cycle

dependabot[bot] committed:
deps: update supranational/blst to 0.3.11 (#27890)

build(deps): bump github.com/supranational/blst

Bumps [github.com/supranational/blst](https://github.com/supranational/blst) from 0.3.11-0.20230406105308-e9dfc5ee724b to 0.3.11.
- [Release notes](https://github.com/supranational/blst/releases)
- [Commits](https://github.com/supranational/blst/commits/v0.3.11)

---
updated-dependencies:
- dependency-name: github.com/supranational/blst
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

imulmat4 committed:
core/txpool/blobpool: fix metrics name for prometheus export (#27901)

jsvisa committed:
ethdb/leveldb: support more than 7 levels in metrics (#27904)

karalabe committed:
all: update golang/x/ext and fix slice sorting fallout (#27909)

The Go authors updated golang/x/ext to change the function signature of the slices sort method. 
It's an entire shitshow now because x/ext is not tagged, so everyone's codebase just 
picked a new version that some other dep depends on, causing our code to fail building.

This PR updates the dep on our code too and does all the refactorings to follow upstream...

fjl committed:
go.mod, build: upgrade c-kzg-4844 (#27907)

This upgrades to the latest release of ckzg, and also attempts to fix some blst-related
build errors that occur on launchpad.net.

fjl committed:
internal/build: apply -ubuntu to env (#27910)

fjl committed:
params: release go-ethereum v1.12.2

meowsbits committed:
wip

Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
params/types/coregeth,params/types/ctypes,params/types/genesisT,params/types/goethereum: install Shanghai=Spiral block number based iface methods

ETC plans to install most EIPs associated with
ETH's Shanghai fork, but ETC doesn't use PoS so activation
by block time, so this adds supporting block-based activation
methods for those features.
Still TODO is to actually implement the associated conditionals
where the Getters are used in the code to actually turn
the new features on.

Date: 2023-08-18 09:11:04-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
params: (typo) remove spurious and meaningless line

Date: 2023-08-18 19:28:32-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
params/types/coregeth: fix feature-block type: uint64/big.Int

Blocks are big ints.
Times are uint64s.

I am 73% certain this covers all cases.

Date: 2023-08-18 19:30:17-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
core,core/vm/runtime,params,tests: GetEIP3651Transition

Date: 2023-08-18 19:50:22-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
core,eth/catalyst: SetEIP3651Transition

Date: 2023-08-19 07:24:31-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
consensus/clique,core/vm: GetEIP3855Transition

Date: 2023-08-19 07:29:40-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
eth/catalyst: SetEIP3855Transition

Date: 2023-08-19 07:43:32-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
cmd/evm/internal/t8ntool,consensus/ethash,core,core/txpool,core/vm,light: GetEIP3860Transition

Date: 2023-08-19 07:53:58-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
eth/catalyst: SetEIP3860Transition

Date: 2023-08-19 07:55:11-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
cmd/evm/internal/t8ntool,consensus/beacon,core,eth/catalyst: GetEIP4895Transition

Date: 2023-08-19 08:22:35-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
params: (typo) remove spurious character '$'

Date: 2023-08-19 08:23:35-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
eth/catalyst: SetEIP4895Transition

Date: 2023-08-19 08:27:25-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
eth/catalyst: SetEIP6049Transition

Date: 2023-08-19 08:30:24-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
core/vm,params: GetEIP4399Transition (noop for ETC)

Date: 2023-08-19 08:54:33-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
params: install Spiral placeholder activation vals

Date: 2023-08-19 09:13:58-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
params/types/goethereum: fix incompatible check; permit passthrough block-based Shanghai config

Failing was TestCheckCompatible, which
converts Geth::CoreGeth configuration types;
CoreGeth's EIP3651 block transition was fatally
inapplicable to Geth's config.

So this patch sidesteps the issue by permitting
the conversion to quietly fail; expecting that users/developers
will be sane enough to configure upgrades with one pattern or the other
but not both.

Date: 2023-08-19 09:17:33-06:00
Signed-off-by: meows <b5c6@protonmail.com>

ziogaschr committed:
Revert "les: remove obsolete code related to PoW header syncing (#27737)"

This reverts commit d4d88f9bce13ca9310bf28f5f26ea9f1915ba90d.

ziogaschr committed:
: Merge branch 'foundation-release/1.12.2' into merge/foundation-release/1.12.2

ziogaschr committed:
params/types/genesisT: add BlobGasUsed and ExcessBlobGas to Genesis

ziogaschr committed:
params/types/goethereum: add VerkleTime to ChainConfig

ziogaschr committed:
params: remove Rinkeby configs

ziogaschr committed:
params,params/types/goethereum: add IsDevMode in goethereum.ChainConfig

ziogaschr committed:
go.mod,go.sum: go mod tidy

ziogaschr committed:
params/types/coregeth,params/types/ctypes,params/types/genesisT,params/types/goethereum: add EIPs 1153 - 5656 - 6780 in chain configurators

ziogaschr committed:
core/vm: use new interface for Selfdestruct in evmc

ziogaschr committed:
consensus/misc/eip4844,core/vm,trie/triedb/pathdb: /s/params/vars

ziogaschr committed:
core/vm,eth/tracers/native,internal/ethapi: add time arg for vm.PrecompiledContractsForConfig

ziogaschr committed:
core: bring back ChainOverrides.OverrideShanghai

ziogaschr committed:
core: /s/params/vars

ziogaschr committed:
core: remove Rinkeby ref

ziogaschr committed:
fixes

ziogaschr committed:
core/txpool/blobpool,core/txpool/legacypool,eth/downloader: 🩹 fixes

ziogaschr committed:
eth,eth/filters,miner: Revert "miner: suspend miner if node is syncing (#27218)"

This reverts commit d4961881d7c92603f591f9cb8c705d00d8cbdfc0.

ziogaschr committed:
miner: suspend miner if node is syncing

At ETC we reverted d496188, and this commit brings back part of the initial commit.

Disables activities while syncing
-  Disable activities (e.g. generate pending state) while node is syncing,
-  Disable empty block submission (but empty block is still kept for payload building),

ref: d496188

ziogaschr committed:
miner: use of correct Transaction object and not of txpool.Transaction

ziogaschr committed:
eth/ethconfig: txpool fixes

ziogaschr committed:
core/txpool/legacypool,eth: mute RemovePendingTransaction api

ziogaschr committed:
eth/catalyst,params/types/coregeth,params/types/ctypes,params/types/goethereum: handle IsDevMode in configurator

ziogaschr committed:
les/catalyst,miner/stress/beacon: wrong number of params in ExecutableDataToBlock

ziogaschr committed:
eth/catalyst,eth,les: 🩹 fixes

ziogaschr committed:
cmd/utils: remove Rinkeby refs

ziogaschr committed:
params/types/genesisT: add *IsDevMode implementations for genesisT.Genesis

ziogaschr committed:
cmd/echainspec: remove Rinkeby refs from echainspec

ziogaschr committed:
tests: fix broken code syntax from conflict resolutions

ziogaschr committed:
cmd/evm/internal/t8ntool,cmd/faucet,tests: 🩹 fixes

ziogaschr committed:
cmd/faucet: fix faucet

meowsbits committed:
tests: add ETC_Spiral test fork config

Date: 2023-08-25 08:55:58-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
Merge branch 'master' into spiral

meowsbits committed:
params/types/coregeth,params/types/goethereum: problem: TestEquivalent_Features fails

Mismatched EIP3651 activations caused an
incompatibility between core-geth and go-ethereum
config types.

This is mostly just a problem of/with the tests
because of the disparate configuration TYPES
now (block vs. time).

This patches fixes the issue by implementing
a logical short-circut equivocating zero values
for time and block fields in core-geth
and go-ethereum configuration configurator
interfaces.

Probably these logical steps will only be followed
by tests or testnets or networks using time OR
block values.

Date: 2023-08-25 14:51:50-06:00
Signed-off-by: meows <b5c6@protonmail.com>

ziogaschr committed:
miner,miner/stress/1559,miner/stress/beacon,miner/stress/ethash,miner: fix interface issues on tests

ziogaschr committed:
cmd/geth,consensus/misc/eip4844,core,eth/ethconfig,graphql,internal/ethapi,params,params/types/genesisT: fix interface changes for tests

ziogaschr committed:
internal/ethapi: internal/ethapi,params/types/genesisT: internal/ethapi: don't return header size from rpc (#27347)

RPC methods `eth_getHeaderBy*` returned a size value which was meant for internal
processes. Please instead use `size` field returned by `eth_getBlockBy*` if you're interested
in the RLP encoded storage size of the block.

ref: 1816cdc

ziogaschr committed:
internal/ethapi: internal/ethapi: internal/ethapi: Add Withdrawal fields in RPCMarshalBlockT and RPCMarshalHeaderT

ziogaschr committed:
internal/ethapi: internal/ethapi: internal/ethapi: bring back totalDifficulty as expected output in TestRPCMarshalBlock

meowsbits committed:
ethclient: assignment mismatch: 2 variables but backend.Attach returns 1 value

Date: 2023-08-28 16:19:53-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
core/txpool/blobpool: undefined: params.BlobTxHashVersion

Date: 2023-08-28 16:35:17-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
core/txpool/blobpool: undefined: params.ChainConfig

Date: 2023-08-28 16:36:04-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
core/txpool/blobpool: undefined: params.InitialBaseFee

Date: 2023-08-28 16:36:24-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
core/txpool/blobpool: use ChainConfigurator types where applicable

Date: 2023-08-28 16:39:37-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
core/txpool/blobpool: undefined: params.BlobTxMinBlobGasprice

Date: 2023-08-28 16:40:02-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
core/txpool/legacypool: not enough arguments in call to newTestBlockChain

Date: 2023-08-28 16:43:19-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
core/types: undefined: params.BlobTxBlobGasPerBlob

Date: 2023-08-28 16:44:06-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
core: assignment mismatch: 2 variables but plot.New returns 1 value

Date: 2023-08-28 16:57:03-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
core: array length vars.MaxInitCodeSize + 1 (value of type uint64) must be constant

Date: 2023-08-28 17:01:19-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
core: undefined: misc

Date: 2023-08-28 17:02:44-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
core/txpool/blobpool: core/txpool/blobpool/evictheap_test.go:189:29: undefined: params.BlobTxBytesPerFieldElement
core/txpool/blobpool/evictheap_test.go:189:65: undefined: params.BlobTxFieldElementsPerBlob
core/txpool/blobpool/evictheap_test.go:249:29: undefined: params.BlobTxBytesPerFieldElement
core/txpool/blobpool/evictheap_test.go:249:65: undefined: params.BlobTxFieldElementsPerBlob (typecheck)

Date: 2023-08-28 17:03:50-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
core/vm: cannot use evmInterpreter (variable of type Interpreter) as *EVMInterpreter value in argument to opBlobHash: need type assertion

Date: 2023-08-28 17:05:21-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
eth/catalyst: undefined: core.Genesis

Date: 2023-08-28 17:07:32-06:00
Signed-off-by: meows <b5c6@protonmail.com>

ziogaschr committed:
core/vm: fix precompiled contracts assignment in tests

ziogaschr committed:
ethclient: remove `debug_setTrieFlushInterval` in supported RPC methods test

ziogaschr committed:
core: fix TestStateProcessorErrors test, because of using wrong consensus

ziogaschr committed:
dev mode replace clique with catalyst for `--dev`

This commit fixes the replacement of clique with catalyst for dev mode on core-geth based on ea782809 .

It still allows use of `—dev.pow`

ziogaschr committed:
eth/filters: eth/filters: eth/filters: fix TestFilters test

ziogaschr committed:
core: warn that Verkle fork is not yet supported

ziogaschr committed:
internal/cmdtest: remove self note

ziogaschr commented on eth/api_debug.go:
```diff
@@ -0,0 +1,417 @@
+// RemovePendingTransaction removes a transaction from the txpool.
+// It returns the transaction removed, if any.
+func (api *DebugAPI) RemovePendingTransaction(hash common.Hash) (*types.Transaction, error) {
+	// TODO (ziogaschr): add support for RemoveTx in bloppool
```
@meowsbits this is a leftover, I will have a look to it. Have you already checked into blobpool?

ziogaschr replied:
We decided to remove support for it.

ziogaschr commented on eth/api_debug.go:
```diff
@@ -0,0 +1,417 @@
+// RemovePendingTransaction removes a transaction from the txpool.
+// It returns the transaction removed, if any.
+func (api *DebugAPI) RemovePendingTransaction(hash common.Hash) (*types.Transaction, error) {
+	// TODO (ziogaschr): add support for RemoveTx in bloppool
```
```suggestion
	// TODO (ziogaschr): add support for RemoveTx in blobpool
```

ziogaschr committed:
Update eth/api_debug.go

ziogaschr committed:
params/types/genesisT: fix indentation

ziogaschr committed:
params/types/genesisT: params/types/genesisT: fix indentation

ziogaschr committed:
tests: tests: set new head for tests/testdata

ziogaschr committed:
tests: set new head for tests/testdata

ziogaschr committed:
tests: tests: Set new submodule head for tests/testdata-etc

ziogaschr committed:
tests: Set new submodule head for tests/testdata-etc

meowsbits committed:
Merge branch 'spiral' into merge/foundation-release/1.12.2+spiral

Conflicts:
      core/genesis.go
      core/txpool/txpool.go
      core/vm/jump_table.go
      params/types/genesisT/genesis.go
      tests/init.go

meowsbits committed:
beacon/engine: include error in wrapped error Error method

By Go conventions the Error method should
return a descriptive string.

When I was debugging the simulated beacon
test, the reported error was hardly usedful,
describing only that an error existed, but lacking
any contextual detail.

Now the engine api will return a more
detailed error and that will hopefully
help debuggers in the future.

Date: 2023-08-29 09:01:11-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
eth/catalyst: fix EIP4895 conditional logic

This logic was just wrong because of
misplaced negation.

Date: 2023-08-29 09:10:51-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
Revert "beacon/engine: include error in wrapped error Error method"

This reverts commit ae0efa109d37e8374999a304bb03dd3ece6bde2a.

Reason: it caused a test failure TestGetBlockBodiesByRangeInvalidParams
where that test compares the have/want errors as strings
and I can't be bothered to adjust the test.

It would be better, probably, to use errors.Is with wrapped
errors.

meowsbits committed:
tests: skip post-Berlin forks during EVMC tests

CI tests were failing because the Cancun
suite was not skipped.

Core-Geth still only supports EVMCv7, so
all forks thereafter need to be skipped.

AFAIK an upgrade to EVMCv10 should catch
up to compatibility for these later forks.

Date: 2023-08-29 09:55:26-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
Merge branch 'merge/foundation-release/1.12.2' into merge/foundation-release/1.12.2+spiral

diega commented on cmd/clef/README.md:
```diff
@@ -29,7 +29,7 @@ GLOBAL OPTIONS:
    --keystore value        Directory for the keystore (default: "$HOME/.ethereum/keystore")
    --configdir value       Directory for Clef configuration (default: "$HOME/.clef")
-   --chainid value         Chain id to use for signing (1=mainnet, 2=Morden (disused), 3=ropsten, 4=rinkeby, 5=Goerli) (default: 1)
+   --chainid value         Chain id to use for signing (1=mainnet, 5=Goerli) (default: 1)
```
```suggestion
   --chainid value         Chain id to use for signing (1=foundation, 5=Goerli, 61=classic, 63=Mordor) (default: 1)
```

ziogaschr committed:
core/txpool/legacypool,eth,eth/tracers,ethclient,internal/web3ext: remove debug_removePendingTransaction

diega commented on cmd/devp2p/README.md:
```diff
@@ -44,7 +44,7 @@ set to standard output. The following filters are supported:
 - `-ip <CIDR>` filters nodes by IP subnet
 - `-min-age <duration>` filters nodes by 'first seen' time
-- `-eth-network <mainnet/rinkeby/goerli/sepolia>` filters nodes by "eth" ENR entry
+- `-eth-network <mainnet/goerli/sepolia>` filters nodes by "eth" ENR entry
```
```suggestion
- `-eth-network <classic/mordor/mainnet/goerli/sepolia>` filters nodes by "eth" ENR entry
```

meowsbits committed:
beacon/engine,eth/catalyst: catalyst api payload attributes get block number

In order to check eip4895 features for ETC-style
block number-based activations, the attributes get
a block number field which is allowed to be nil.

As usual for core-geth, activation numbers are OR'd
with activation times.

Date: 2023-09-01 10:18:47-06:00
Signed-off-by: meows <b5c6@protonmail.com>

diega reviewed (approved):
This looks really well. Just made two microscopical comments

ziogaschr committed:
Update cmd/clef/README.md

Co-authored-by: Diego López León <dieguitoll@gmail.com>

ziogaschr committed:
Update cmd/devp2p/README.md

Co-authored-by: Diego López León <dieguitoll@gmail.com>

meowsbits committed:
core: prefer EIP descriptors over fork names

... but reference forks by name to
keep the important context.

Date: 2023-09-04 05:22:48-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
docs/JSON-RPC-API/modules: make docs-generate

Date: 2023-09-04 08:03:21-06:00
Signed-off-by: meows <b5c6@protonmail.com>

meowsbits committed:
params/types/genesisT: omitempty tags for optional genesis fields

These fields are not supported by all
chains which core-geth supports.
Since chain config data types omit empty values,
it follows that genesis ought follow suit.

Date: 2023-09-04 08:38:51-06:00
Signed-off-by: meows <b5c6@protonmail.com>

ziogaschr committed:
params/types/coregeth,params/types/goethereum: Revert "params/types/coregeth,params/types/goethereum: problem: TestEquivalent_Features fails"

This reverts commit a5e0c62fe940d93578105b8c8cd81688cd13f174.

ziogaschr committed:
params/types/coregeth: fix typo on core-geth chain configurator EIPs for Shanghai

meowsbits committed:
params: unset Spiral/Shanghai activation blocks for Classic, Mordor

The ECIP does not specify valid activation
blocks yet, so we should not code them.
https://github.com/ethereumclassic/ECIPs/blob/64423c73b7416c2f8e40d4c66bf40b715c458b8c/_specs/ecip-1109.md

Date: 2023-09-04 10:34:38-06:00
Signed-off-by: meows <b5c6@protonmail.com>

ziogaschr committed:
core/forkid: unset Spiral activation blocks tests for Classic

meowsbits committed:
Merge pull request #561 from etclabscore/merge/foundation-release/1.12.2+spiral

Merge/foundation release/1.12.2+spiral
